modules/geolocation/geolocation.py

```diff
@@ -96,6 +95,81 @@ def gather_point_pairs(self):
             pixelGeoPairs = np.concatenate((pixelGeoPairs, [pair]))
 
         return pixelGeoPairs
+    
+    def __are_three_points_collinear(self, p1, p2, p3):
+        """
+        PRIVATE
+        Evaluates whether the three given points are collinear
+
+        Parameters
+        ----------
+        p1 : tuple/np.ndarray/list
+            Point 1, in form (x, y)
+        p2 : tuple/np.ndarray/list
+            Point 2, in form (x, y)
+        p3 : tuple/np.ndarray/list
+            Point 3, in form (x, y)
+
+        Returns
+        -------
+        bool
+            True if the three points are collinear, otherwise false
+        """
+        x = 0
+        y = 1
+        # Calculates the area of a triangle and checks if this value is 0
+        # Actually calculates 2 times the area, since it skips the unnecessary step of multiplication by 0.5
+        return (p1[x]*(p2[y]-p3[y]) +
+                p2[x]*(p3[y]-p1[y]) +
+                p3[x]*(p1[y]-p2[y])) == 0
+
+
+    def get_non_collinear_points(self, coordinatesArray):
+        """
+        Returns a list of four coordinates from an input array that are not collinear to one another.
+
+        Parameters
+        ----------
+        coordinatesArray : np.ndarray
+            Array with dimensions ( , 2), containing a list of coordinates
+
+        Returns
+        -------
+        np.ndarray
+            Array with dimensions (4, 2), containing a list of coordinates that are non-collinear,
+            or an empty list if none were found in the input array
+        """
+        NUM_POINTS_NEEDED = 4
+        # Empty array in case no set of four non-collinear points are found
+        collinearPoints = np.empty(shape=(0, 2))
+
+        # If there aren't four points, return the empty array
+        if len(coordinatesArray) < NUM_POINTS_NEEDED:
+            return collinearPoints
```

Comment:
collinearPoints is just an array of size 0, why not `return np.empty(shape=(0, 2))` directly?